### PR TITLE
[NCL-2097] removed javax.xml.bind.annotations

### PR DESCRIPTION
**Pull request by stanlyDoge**

### CODE REVIEW

File: bc-model/src/test/java/org/jboss/da/bc/model/rest/BackwardCompatibilityTest.java [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.junit.Assert.fail;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.junit.Assert.fail;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.junit.Assert.fail;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.junit.Assert.fail;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.junit.Assert.fail;
```
Comment (janinko) [inline bc-model/src/test/java/org/jboss/da/bc/model/rest/BackwardCompatibilityTest.java]: What is this? :scream:

File: bc-model/src/test/resources/backwardCompatibilityTest/BuildConfiguration.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"analysisStatus":"ANALYSED","scmUrl":"scmUrl","scmRevision":"scmRevision","cloneRepo":true,"projectId":0,"buildScript":"buildScript","name":"name","description":"description","internallyBuilt":"internallyBuilt","availableVersions":[],"bcId":0,"existingBCs":[],"gav":{"groupId":"g","artifactId":"a","version":"v"},"environmentId":1,"dependencies":[],"selected":true,"errors":["No dependency found:Failed to found dependencies in Aprox and SCM repository","Enviroment ID is null:There is no enviroment selected or another error occured and enviroment Id is null"]}
+
```
Comment (janinko) [inline bc-model/src/test/resources/backwardCompatibilityTest/BuildConfiguration.json]: Would it be possible to have all this jsons formated?

File: communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java [hunk 3]
```diff
@@ -221,10 +221,17 @@ public boolean doesGAVExistInPublicRepo(GAV gav) throws CommunicationException {
...
+        return s.hasNext() ? s.next() : "";
+    }
+
     private VersionResponse parseMetadataFile(URLConnection connection) throws IOException,
             CommunicationException {
         try (InputStream in = connection.getInputStream()) {
-            return MetadataFileParser.parseMetadataFile(in);
+            //System.out.println("inputstream " + convertStreamToString(in));
+            VersionResponse k = MetadataFileParser.parseMetadataFile(in);
+            return k;
```
Comment (janinko) [inline communication/src/main/java/org/jboss/da/communication/aprox/impl/AproxConnectorImpl.java]: Forgotten debug?

File: reports-backend/src/test/java/org/jboss/da/reports/api/BackwardCompatibilityTest.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        private String convertPath(String path) {
+            String convertedPath = path.endsWith("/") ? path.substring(0, path.length() - 1)
+                    + "index" : path;
+            return convertedPath.startsWith("/") ? convertedPath.substring(1) : convertedPath;
+        }
+
+        private String convertContentType(ContentType contentType) {
+            if ("text/html".equals(contentType.getMimeType()))
+                return ".html";
```
Comment (janinko) [inline reports-backend/src/test/java/org/jboss/da/reports/api/BackwardCompatibilityTest.java]: You are doing something with HTML files?

File: reports-model/src/main/java/org/jboss/da/model/rest/GAV.java [hunk 5]
```diff
@@ -30,9 +28,9 @@ public GAV(@JsonProperty("groupId") String groupId,
     }
 
     @JsonIgnore
-    @XmlTransient
     /*
-     * The @XmlTransient annotation is added so that Swagger doesn't try to represent the 'GA' object in its model schema
+     * @XmlTransient The @XmlTransient annotation is added so that Swagger doesn't try to represent the 'GA' object in its model
+     * schema
```
Comment (janinko) [inline reports-model/src/main/java/org/jboss/da/model/rest/GAV.java]: Why the comment still when there is no @XmlTransient?

File: reports-model/src/main/java/org/jboss/da/products/model/rest/GADiff.java [hunk 6]
```diff
@@ -46,7 +46,7 @@ public GADiff(GA ga, String leftVersion, String rightVersion) {
     }
 
     @JsonIgnore
-    @XmlTransient
+    //@XmlTransient
```
Comment (janinko) [inline reports-model/src/main/java/org/jboss/da/products/model/rest/GADiff.java]: Why it's commented out and not removed?

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by stanlyDoge** [comment]
retest this please

**Comment by janinko** [comment]
> Would it be possible to have all this jsons formated?

Please format all of them.

**Comment by janinko** [comment]
I think that ``org.apache.http.entity.ContentType;`` is not actually used, so it can be removed and also the ``org.apache.httpcomponents:httpcore`` dependency can be removed.

**Comment by janinko** [comment]
Reworked in  #438

**Pull request closed without merging**